domain/application/types.go

```diff
@@ -22,6 +23,16 @@ type AddApplicationArg struct {
 	Scale             int
 	Platform          Platform
 	Channel           *Channel
+	// Resources defines the list of resources to add to an application.
+	// They should match all the resources defined in the Charm.
+	Resources []AddApplicationResourceArg
+}
+
+// AddApplicationResourceArg defines the arguments required to add a resource to an application.
+type AddApplicationResourceArg struct {
```

Comment:
Just to check, is this information all readily available when the application is created?

Reply:
Yes it should. It is basically the merge between two source of information:

* charm metadata
* command line args.